<!doctype html><html lang=en><head><meta charset=utf-8><meta name=description content="GitHub:: https://gitlab.com/wtsi-grit/rapid-curation
Contribution:: Curation of a reference quality genome.
Working with:: AT, JMDW, KK, MUS
Topic:: Data driven decision making, team work"><title>Curation of iyTipFemo1</title><meta name=viewport content="width=device-width,initial-scale=1"><link rel="shortcut icon" type=image/png href=https://dlbpointon.github.io/TDB-obs-quartz//icon.png><link href=https://dlbpointon.github.io/TDB-obs-quartz/styles.cdff8e4a99b7e2d68e511c19d45a07cb.min.css rel=stylesheet><link href=https://dlbpointon.github.io/TDB-obs-quartz/styles/_light_syntax.86a48a52faebeaaf42158b72922b1c90.min.css rel=stylesheet id=theme-link><script src=https://dlbpointon.github.io/TDB-obs-quartz/js/darkmode.cfcd25279560061512e85b624dd83073.min.js></script>
<script src=https://dlbpointon.github.io/TDB-obs-quartz/js/util.5e39932758f9ecaf45fd54506ed61416.min.js></script>
<link rel=stylesheet href=https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.css integrity=sha384-R4558gYOUz8mP9YWpZJjofhk+zx0AS11p36HnD2ZKj/6JR5z27gSSULCNHIRReVs crossorigin=anonymous><script defer src=https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.js integrity=sha384-z1fJDqw8ZApjGO3/unPWUPsIymfsJmyrDVWC8Tv/a1HeOtGmkwNd/7xUS0Xcnvsx crossorigin=anonymous></script>
<script defer src=https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/contrib/auto-render.min.js integrity=sha384-+XBljXPPiv+OzfbB3cVmLHf4hdUFHlWNZN5spNQ7rmHTXpd7WvJum6fIACpNNfIR crossorigin=anonymous></script>
<script defer src=https://cdn.jsdelivr.net/npm/katex@0.16.2/dist/contrib/copy-tex.min.js integrity=sha384-ww/583aHhxWkz5DEVn6OKtNiIaLi2iBRNZXfJRiY1Ai7tnJ9UXpEsyvOITVpTl4A crossorigin=anonymous></script>
<script src=https://unpkg.com/@floating-ui/core@0.7.3></script>
<script src=https://unpkg.com/@floating-ui/dom@0.5.4></script>
<script src=https://dlbpointon.github.io/TDB-obs-quartz/js/popover.6da9b273c092cc16fc1aa904d71a2163.min.js></script>
<script src=https://dlbpointon.github.io/TDB-obs-quartz/js/code-title.b35124ad8db0ba37162b886afb711cbc.min.js></script>
<script src=https://dlbpointon.github.io/TDB-obs-quartz/js/clipboard.c20857734e53a3fb733b7443879efa61.min.js></script>
<script src=https://dlbpointon.github.io/TDB-obs-quartz/js/callouts.7723cac461d613d118ee8bb8216b9838.min.js></script>
<script>const SEARCH_ENABLED=!1,LATEX_ENABLED=!0,PRODUCTION=!0,BASE_URL="https://dlbpointon.github.io/TDB-obs-quartz/",fetchData=Promise.all([fetch("https://dlbpointon.github.io/TDB-obs-quartz/indices/linkIndex.43525d156f7c2dfb16a0273a95d808be.min.json").then(e=>e.json()).then(e=>({index:e.index,links:e.links})),fetch("https://dlbpointon.github.io/TDB-obs-quartz/indices/contentIndex.12f57070e5e1faeb2f632e94dc7ed39e.min.json").then(e=>e.json())]).then(([{index:e,links:t},n])=>({index:e,links:t,content:n})),render=()=>{const e=new URL(BASE_URL),t=e.pathname,n=window.location.pathname,s=t==n;addCopyButtons(),addTitleToCodeBlocks(),addCollapsibleCallouts(),initPopover("https://dlbpointon.github.io/TDB-obs-quartz",!0);const o=document.getElementById("footer");if(o){const e=document.getElementById("graph-container");if(!e)return requestAnimationFrame(render);e.textContent="";const t=s&&!1;drawGraph("https://dlbpointon.github.io/TDB-obs-quartz",t,[{"/moc":"#4388cc"}],t?{centerForce:1,depth:-1,enableDrag:!0,enableLegend:!1,enableZoom:!0,fontSize:.5,linkDistance:1,opacityScale:3,repelForce:1,scale:1.4}:{centerForce:1,depth:1,enableDrag:!0,enableLegend:!1,enableZoom:!0,fontSize:.6,linkDistance:1,opacityScale:3,repelForce:2,scale:1.2})}var i=document.getElementsByClassName("mermaid");i.length>0&&import("https://unpkg.com/mermaid@9/dist/mermaid.esm.min.mjs").then(e=>{e.default.init()})},init=(e=document)=>{addCopyButtons(),addTitleToCodeBlocks(),renderMathInElement(e.body,{delimiters:[{left:"$$",right:"$$",display:!0},{left:"$",right:"$",display:!1}],macros:{'’':"'"},throwOnError:!1})}</script><script type=module>
    import { attachSPARouting } from "https:\/\/dlbpointon.github.io\/TDB-obs-quartz\/js\/router.9d4974281069e9ebb189f642ae1e3ca2.min.js"
    attachSPARouting(init, render)
  </script></head><body><div id=search-container><div id=search-space><input autocomplete=off id=search-bar name=search type=text aria-label=Search placeholder="Search for something..."><div id=results-container></div></div></div><script src=https://cdn.jsdelivr.net/npm/flexsearch@0.7.21/dist/flexsearch.bundle.js integrity="sha256-i3A0NZGkhsKjVMzFxv3ksk0DZh3aXqu0l49Bbh0MdjE=" crossorigin=anonymous defer></script>
<script defer src=https://dlbpointon.github.io/TDB-obs-quartz/js/full-text-search.51f0b1753e9b30839d053f8a98cc20d1.min.js></script><div class=singlePage><header><h1 id=page-title><a href=https://dlbpointon.github.io/TDB-obs-quartz/>The Dyslexic Bioinformatician</a></h1><div class=spacer></div><div id=search-icon><p>Search</p><svg tabindex="0" aria-labelledby="title desc" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 19.9 19.7"><title id="title">Search Icon</title><desc id="desc">Icon to open search</desc><g class="search-path" fill="none"><path stroke-linecap="square" d="M18.5 18.3l-5.4-5.4"/><circle cx="8" cy="8" r="7"/></g></svg></div><div class=darkmode><input class=toggle id=darkmode-toggle type=checkbox tabindex=-1>
<label id=toggle-label-light for=darkmode-toggle tabindex=-1><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" id="dayIcon" viewBox="0 0 35 35" style="enable-background:new 0 0 35 35"><title>Light Mode</title><path d="M6 17.5C6 16.672 5.328 16 4.5 16h-3C.672 16 0 16.672.0 17.5S.672 19 1.5 19h3C5.328 19 6 18.328 6 17.5zM7.5 26c-.414.0-.789.168-1.061.439l-2 2C4.168 28.711 4 29.086 4 29.5 4 30.328 4.671 31 5.5 31c.414.0.789-.168 1.06-.44l2-2C8.832 28.289 9 27.914 9 27.5 9 26.672 8.329 26 7.5 26zm10-20C18.329 6 19 5.328 19 4.5v-3C19 .672 18.329.0 17.5.0S16 .672 16 1.5v3C16 5.328 16.671 6 17.5 6zm10 3c.414.0.789-.168 1.06-.439l2-2C30.832 6.289 31 5.914 31 5.5 31 4.672 30.329 4 29.5 4c-.414.0-.789.168-1.061.44l-2 2C26.168 6.711 26 7.086 26 7.5 26 8.328 26.671 9 27.5 9zM6.439 8.561C6.711 8.832 7.086 9 7.5 9 8.328 9 9 8.328 9 7.5c0-.414-.168-.789-.439-1.061l-2-2C6.289 4.168 5.914 4 5.5 4 4.672 4 4 4.672 4 5.5c0 .414.168.789.439 1.06l2 2.001zM33.5 16h-3c-.828.0-1.5.672-1.5 1.5s.672 1.5 1.5 1.5h3c.828.0 1.5-.672 1.5-1.5S34.328 16 33.5 16zM28.561 26.439C28.289 26.168 27.914 26 27.5 26c-.828.0-1.5.672-1.5 1.5.0.414.168.789.439 1.06l2 2C28.711 30.832 29.086 31 29.5 31c.828.0 1.5-.672 1.5-1.5.0-.414-.168-.789-.439-1.061l-2-2zM17.5 29c-.829.0-1.5.672-1.5 1.5v3c0 .828.671 1.5 1.5 1.5s1.5-.672 1.5-1.5v-3C19 29.672 18.329 29 17.5 29zm0-22C11.71 7 7 11.71 7 17.5S11.71 28 17.5 28 28 23.29 28 17.5 23.29 7 17.5 7zm0 18c-4.136.0-7.5-3.364-7.5-7.5s3.364-7.5 7.5-7.5 7.5 3.364 7.5 7.5S21.636 25 17.5 25z"/></svg></label><label id=toggle-label-dark for=darkmode-toggle tabindex=-1><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" id="nightIcon" viewBox="0 0 100 100" style="enable-background='new 0 0 100 100'"><title>Dark Mode</title><path d="M96.76 66.458c-.853-.852-2.15-1.064-3.23-.534-6.063 2.991-12.858 4.571-19.655 4.571C62.022 70.495 50.88 65.88 42.5 57.5 29.043 44.043 25.658 23.536 34.076 6.47c.532-1.08.318-2.379-.534-3.23-.851-.852-2.15-1.064-3.23-.534-4.918 2.427-9.375 5.619-13.246 9.491-9.447 9.447-14.65 22.008-14.65 35.369.0 13.36 5.203 25.921 14.65 35.368s22.008 14.65 35.368 14.65c13.361.0 25.921-5.203 35.369-14.65 3.872-3.871 7.064-8.328 9.491-13.246C97.826 68.608 97.611 67.309 96.76 66.458z"/></svg></label></div></header><article><h1>Curation of iyTipFemo1</h1><p class=meta>Last updated
Jan 5, 2023
<a href=https://github.com/DLBPointon/TDB-obs-quartz/Work-based%20Projects/Curation%20of%20iyTipFemo1.md rel=noopener>Edit Source</a></p><ul class=tags><li><a href=https://dlbpointon.github.io/TDB-obs-quartz/tags/TDB/>Tdb</a></li><li><a href=https://dlbpointon.github.io/TDB-obs-quartz/tags/workbased/>Workbased</a></li><li><a href=https://dlbpointon.github.io/TDB-obs-quartz/tags/portfolio/>Portfolio</a></li></ul><aside class=mainTOC><details><summary>Table of Contents</summary><nav id=TableOfContents><ol><li><ol><li><a href=#the-project>The Project</a></li><li><a href=#my-contribution>My Contribution</a></li><li><a href=#future-plans>Future Plans</a></li><li><a href=#references>References</a></li></ol></li></ol></nav></details></aside><p>GitHub::
<a href=https://gitlab.com/wtsi-grit/rapid-curation rel=noopener>https://gitlab.com/wtsi-grit/rapid-curation</a></p><p>Contribution:: Curation of a reference quality genome.</p><p>Working with:: AT, JMDW, KK, MUS</p><p>Topic:: Data driven decision making, team work</p><p>KSBs:: K5-5, K3-3, S6, S7, B3</p><hr><p>Description:: How I have curated the reference genome for <em>Tiphia femorata</em>.</p><a href=#the-project><h3 id=the-project><span class=hanchor arialabel=Anchor># </span>The Project</h3></a><p>The aims of the DTOL project include the production of reference quality genomes for the 77,000 organisms that call the British Isles and Ireland their home. A part of this mamoth task is ensuring that practically anyone can perform such a task with only some help from the professional curators in GRIT.</p><p>This portfolio peice will focus on the curation attempt of <em>Tiphia femorata</em>, known internally as iyTipFemo1 or commonly as the beetle-killing wasp (as it is the parasitoid of the scarab beetle), see Figure 1.</p><p><a class="internal-link broken">Tiphia_femorata.png</a>
Figure 1: A female <em>Tiphia femorata</em>, commonly known as the beetle-killing wasp. Picute taken by Hectonichus (2010).</p><a href=#my-contribution><h3 id=my-contribution><span class=hanchor arialabel=Anchor># </span>My Contribution</h3></a><p>The data received by myself and the other members of ToLa is &ldquo;pre-assembled&rdquo;, meaning that automated pipelines have already taken the raw read data from the Sequencing Team and generated a best-guess assembly. This assembly is then evaluated by those in ToLa with the help of
<a href=https://tolqc.cog.sanger.ac.uk/darwin/insects/Tiphia_femorata/ rel=noopener>TolQC</a>, a website produced to collate assembly statistics as well as preliminary contamination reports and includes graphs as shown in Figure 2 and 3.</p><p><a class="internal-link broken">iyTipFemo1-kmerplot.png</a>
FIgure 2: A Kmer multiplicity plot, that highlights a fairly well assembled diploid genome&mdash;e.g. the first red peak is >50% of the larger peak and the grey peak being kmers that have been removed but have no impact on the quality of the genome</p><p>The iyTipFemo1 assembly it self was unremarkable&mdash;apart from two noticable regions of highly repetitive regions&mdash;and the size of the genome (~300MBp) was small enough that it assembled well via the automated processes, see Figure 3. This isn&rsquo;t a perfect process and misjoins (where contigs that do not actually need joining are joined by the assembler) may need correcting with a break (the breaking of a scaffold into it&rsquo;s component contigs). BUSCO was also run, as standard, on the reads for this assembly resulting in the scores found in Table 1.</p><table><thead><tr><th>Field</th><th>Score (%)</th></tr></thead><tbody><tr><td>Completeness</td><td>99.5%</td></tr><tr><td>Single Copy</td><td>99.1%</td></tr><tr><td>Duplicate</td><td>0.4%</td></tr><tr><td>Fragmented</td><td>0.1%</td></tr><tr><td>Missing</td><td>0.4%</td></tr><tr><td>Table 1: The BUSCO summary results showing that the assembly is estimated to be 99.5% complete with another 0.1% of sequnce being fragmented BUSCO genes. This analysis was performed across the 1367 reads that make up the iyTipFemo1 assembly.</td><td></td></tr></tbody></table><p>The mitochrondial genome required more investigation. My collegue, MUS, developed a python based workflow called
<a href=https://github.com/marcelauliano/MitoHiFi rel=noopener>MitoHiFi</a>, in order to tease out putative mitochondrial reads (HiFi data) via the annotation and whether or not they circularise. In the case of <em>T. femorata</em> multiple potential mitochondrial sequences were identified, an issue that has become rather common particularly in wasp genomes.</p><p><a class="internal-link broken">iytipfemo-precuration.png</a>
Figure 3: The
<a href=https://github.com/wtsi-hpag/PretextView rel=noopener>PretextView</a> of the iyTipFemo1 assembly pre-curation.</p><p>In order to narrow our search (myself, KK and MUS at this point) we employed the use of MBG (Rautiainen, 2021), CD-HIT (Fu, 2012) and Bandage (Wick, 2015). Using CD-HIT, we were able to cluster together scaffolds which were similar to a reference (a mito previously found in a wasp, determined by MitoHiFi) which resulted in two scaffolds being identified as potential mitochondrial reads, ptg00002l and ptg00003. Using MitoHiFi, MBG and Bandage it was then possible to visualise these and see their length and number of genes annotated, this required the use of multiple re-runs, modifying variables such as the clustering cut-offs in CD-HIT to ge the most optimal results. ptg00003 was then assigned as the mitochondrial sequence due to its completeness&mdash;e.g. it was a standard mitochondrial length (16kbp), carried the correct number of genes (37) and was circular&mdash;the other sequence may have been neomite (mitochondrial sequence adopted into the host genome) or a product of heteroplasmy.</p><table><thead><tr><th>Scaffold A</th><th>Scaffold B</th></tr></thead><tbody><tr><td>scaffold_14.13</td><td>scaffold_1.65</td></tr><tr><td>scaffold_2.1</td><td>scaffold_2.20_2</td></tr><tr><td>scaffold_5.49</td><td>scaffold_5.1</td></tr><tr><td>scaffold_6.1</td><td>scaffold_6.20_2</td></tr><tr><td>scaffold_15.1</td><td>scaffold_10.28</td></tr><tr><td>scaffold_16.4_1</td><td>scaffold_11.26</td></tr><tr><td>scaffold_9.1</td><td>scaffold_12.1</td></tr><tr><td>Table 2: A table showing the scaffolds which were joined together manually. Scaffold id&rsquo;s ending with a &lsquo;_1&rsquo; or &lsquo;_2&rsquo; are the products of broken scaffold</td><td></td></tr></tbody></table><table><thead><tr><th>Scaffold broken</th><th>At base pair position</th></tr></thead><tbody><tr><td>scaffold_2.20</td><td>9776444</td></tr><tr><td>scaffold_5.33</td><td>16411527</td></tr><tr><td>scaffold_6.20</td><td>9954728</td></tr><tr><td>scaffold_16.4</td><td>1629000</td></tr><tr><td>Table 3: A table showing the scaffolds which required breaking and the base pair position where the break was introduced.</td><td></td></tr></tbody></table><p>The curation of this assembly was relatively straight forward, requiring seven scaffold joins (see Table 2), four scaffold breaks (see Table 3), and one haplotypic sequence removal (these are moved into the assemblies haplotype file). These were all evidenced by the PretextView map (See figure 3 and 4) and HiGlass map (no longer availble due to size constraints). These contact maps make up the back bone of modern curation, allowing for high resolution interigation of HiC data (HiGlass) as well as for fast and easy manipulation of the assembly in a visual environment (PretextView).</p><p><a class="internal-link broken">iytipfemo-postcuration.png</a>
Figure 4: The
<a href=https://github.com/wtsi-hpag/PretextView rel=noopener>PretextView</a> for the iyTipFemo1 assembly, post-curation. Centromeres, if found, have also been orriented to the left of their respective chromosome.</p><p>Once these changes have been made, the PretextView is then sent to another curator for verification. If they agree that your changes were correct then the curation is complete and the last steps of GRIT involve the submittion of the final genomic fasta file for upload to NCBI. In the case of <em>T. femorata</em> this is
<a href="https://www.ncbi.nlm.nih.gov/genome/?term=txid330862[Organism:noexp]" rel=noopener>here</a> and statistics are found on
<a href=https://grit-realtime.tol.sanger.ac.uk/table.html rel=noopener>GRIT-realtime</a>.</p><p><a class="internal-link broken">iyTipFemo-nucmer.png</a>
Figure 5: A Nucmer plot of the pre-curation iyTipFemo1 assembly against the <em>Orussus abietinus</em> (a sawfly, basal to wasps and bees) assembly.</p><p>The process of manually curating a genome is iterative, relying on evidence (such as the contact maps or syntenic alignments such as those produced with Nucmer, see figure 5) and can sometimes feel like playing Soduko. In any place you may need to make multiple moves and changes based on a educated guesses and these changes may be genuine or may need completely reverting.</p><p>As curation is not my typical role, this project was conducted in order to learn the whole process of assembly through the curation, this required a degree of time management whilst understanding that the curation had to be performed in a reasonable amount in order to keep the pipelines flowing. Due to this as well as time constraints with a curation colleuge, it was decided that Tuesdays and Thursdays would be &ldquo;in contact&rdquo; days where we would talk about changes needed and about potential exploratory work.</p><a href=#future-plans><h3 id=future-plans><span class=hanchor arialabel=Anchor># </span>Future Plans</h3></a><p>I plan on furthering my skills in genomic curation by taking on more tickets in GRIT. Allowing my to get a greater grasp of the shear complexity of genomes through the Hymenopteran genomes as well as the how this compares across the three of life.</p><a href=#references><h3 id=references><span class=hanchor arialabel=Anchor># </span>References</h3></a><p>Fu, L., Niu, B., Zhu, Z., Wu, S., Li, W. 2012. CD-HIT: accelerated for clustering the next-generation sequencing data. <em>Bioinformatics</em>. 28(23). pp. 3150–3152. DOI:
<a href=https://doi.org/10.1093%2Fbioinformatics%2Fbts565 rel=noopener>10.1093/bioinformatics/bts565</a></p><p>Hectonichus. 2010. Tiphiidae - Tiphia femorata. <em>Wikipedia</em>.
<a href=https://en.wikipedia.org/wiki/Tiphia_femorata#/media/File:Tiphiidae_-_Tiphia_femorata..jpg rel=noopener>https://en.wikipedia.org/wiki/Tiphia_femorata#/media/File:Tiphiidae_-_Tiphia_femorata..jpg</a>.</p><p>Rautiainen, M., Marschall, T. 2021. MBG: Minimizer-based sparse de Bruijn Graph construction. <em>Bioinformatics</em>. 37(16). pp. 2476–2478. DOI:
<a href=https://doi.org/10.1093/bioinformatics/btab004 rel=noopener>10.1093/bioinformatics/btab004</a>.</p><p>Wick, RR., Schultz, MB., Zobel, J., Holt, KE. 2015. Bandage: interactive visualization of de novo genome assemblies. <em>Bioinformatics</em>. 31(20). pp. 3350–3352. DOI:
<a href=https://doi.org/10.1093/bioinformatics/btv383 rel=noopener>10.1093/bioinformatics/btv383</a></p></article><hr><div class=page-end id=footer><div class=backlinks-container><h3>Backlinks</h3><ul class=backlinks><li><a href=/TDB-obs-quartz/Computer-Science/University-and-CS-Homepage/ data-ctx="Curation of iyTipFemo1" data-src=/Computer-Science/University-and-CS-Homepage class=internal-link></a></li><li><a href=/TDB-obs-quartz/Portfolio-Pieces/Data-driven-decision-making/ data-ctx="Work-based Projects/Curation of iyTipFemo1" data-src=/Portfolio-Pieces/Data-driven-decision-making class=internal-link>Data driven decision making</a></li><li><a href=/TDB-obs-quartz/Work-based-Projects/Curation-of-idCriBerb1/ data-ctx="Work-based Projects/Curation of iyTipFemo1" data-src=/Work-based-Projects/Curation-of-idCriBerb1 class=internal-link>Curation of idCriBerb1</a></li><li><a href=/TDB-obs-quartz/Work-based-Projects/Work-based-Project-Homepage/ data-ctx="Curation of iyTipFemo1" data-src=/Work-based-Projects/Work-based-Project-Homepage class=internal-link></a></li></ul></div><div><script src=https://cdn.jsdelivr.net/npm/d3@6.7.0/dist/d3.min.js integrity="sha256-+7jaYCp29O1JusNWHaYtgUn6EhuP0VaFuswhNV06MyI=" crossorigin=anonymous></script><h3>Interactive Graph</h3><div id=graph-container></div><style>:root{--g-node:var(--secondary);--g-node-active:var(--primary);--g-node-inactive:var(--visited);--g-link:var(--outlinegray);--g-link-active:#5a7282}</style><script src=https://dlbpointon.github.io/TDB-obs-quartz/js/graph.abd4bc2af3869a96524d7d23b76152c7.js></script></div></div><div id=contact_buttons><footer><p>Made by Damon-Lee Pointon using <a href=https://github.com/jackyzha0/quartz>Quartz</a>, © 2023</p><ul><li><a href=https://dlbpointon.github.io/TDB-obs-quartz/>Home</a></li><li><a href=https://twitter.com/DLBPointon>Twitter</a></li><li><a href=https://github.com/DLBPointon>Github</a></li></ul></footer></div></div></body></html>